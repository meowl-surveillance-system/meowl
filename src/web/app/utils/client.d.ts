import cassandra from 'cassandra-driver';
export declare const client: cassandra.Client;
