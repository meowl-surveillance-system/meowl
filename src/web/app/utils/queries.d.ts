export declare const INSERT_USERSID = "INSERT INTO users_id (user_id, email, username, password, sid) VALUES (?, ?, ?, ?, ?)";
export declare const INSERT_USERSNAME = "INSERT INTO users_name (user_id, email, username, password, sid) VALUES (?, ?, ?, ?, ?)";
export declare const SELECT_USERSID_USERID_PASSWORD = "SELECT user_id, password FROM users_name WHERE username = ?";
export declare const UPDATE_USERSID_SID = "UPDATE users_id SET sid = ? WHERE user_id = ?";
export declare const UPDATE_USERSNAME_SID = "UPDATE users_name SET sid = ? WHERE username = ?";
export declare const SELECT_USERSNAME_SID = "SELECT sid FROM users_id WHERE user_id = ?";
export declare const SELECT_SID_SESSION = "SELECT * FROM sessions WHERE sid = ?";
export declare const SELECT_USERSNAME_USERID = "SELECT user_id FROM users_name WHERE username = ?";
export declare const SELECT_CAMERAID_USERID = "SELECT camera_id FROM user_cameras WHERE user_id = ?";
export declare const SELECT_USERID_CAMERAID = "SELECT user_id FROM camera_users WHERE camera_id = ?";
export declare const INSERT_CAMERAID_USERID = "INSERT INTO user_cameras (user_id, camera_id) VALUES (?, ?)";
export declare const INSERT_USERID_CAMERAID = "INSERT INTO camera_users (user_id, camera_id) VALUES (?, ?)";
export declare const INSERT_CAMERAID_STREAMID = "INSERT INTO camera_streams (camera_id, stream_id, stream_date) VALUES(?, ?, ?)";
export declare const SELECT_CAMERAID_STREAMID = "SELECT stream_id FROM camera_streams WHERE camera_id = ?";
export declare const SELECT_STREAMID_METADATA = "SELECT stream_id FROM metadata WHERE stream_id = ?";
export declare const SELECT_CAMERAID_STREAMID_SINGLE = "SELECT stream_id FROM camera_streams WHERE camera_id = ? LIMIT 1";
export declare const INSERT_CAMERAID_LIVE = "INSERT INTO live_cameras (camera_id, live) VALUES (?, ?)";
export declare const SELECT_CAMERAID = "SELECT DISTINCT camera_id FROM camera_streams";
export declare const SELECT_LIVE_CAMERAID = "SELECT live FROM live_cameras WHERE camera_id = ?";
