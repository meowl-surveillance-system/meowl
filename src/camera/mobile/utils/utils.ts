import AsyncStorage from '@react-native-community/async-storage';
/**
 * Makes a request to log out from the server
 * @param serverUrl - Server URL that is hosting the requests
 * @param logoutEndpoint - Endpoint where the logout method is called
 * @returns the response to log out from the server
 */
export async function logout(
  serverUrl: string,
  logoutEndpoint = '/auth/logout',
) {
  const logoutUrl: string = serverUrl + logoutEndpoint;
  console.log(logoutUrl);
  try {
    const logoutResponse = await fetch(logoutUrl, {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        credentials: 'include',
      },
    });
    return logoutResponse;
  } catch (err) {
    return null;
  }
}

/**
 * Makes a request to log into the server
 * @param credentials - Credentials required to login (e.g. username/password)
 * @param serverUrl - Server URL that is hosting the requests
 * @param loginEndpoint - Endpoint where the login method is called
 * @returns the response to log into the server
 */
export async function login(
  credentials: object,
  serverUrl: string,
  loginEndpoint = '/auth/login',
) {
  const loginUrl = serverUrl + loginEndpoint;
  try {
    const loginResponse = await fetch(loginUrl, {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        credentials: 'include',
      },
      body: JSON.stringify(credentials),
    });
    return loginResponse;
  } catch (err) {
    return null;
  }
}

/**
 * Get item from AsyncStorage if found else return value from args
 * 
 * @param key - Key to retrieve value from AsyncStorage
 * @param value - value to be used if value not found in AsyncStorage
 * @returns value
 */
export async function getItemFromStore(
  key: string,
  value: string = ''
): Promise<string> {
  const valueFromStore: string | null = await AsyncStorage.getItem(key);
  if (valueFromStore) {
    return valueFromStore;
  }
  return value;
}