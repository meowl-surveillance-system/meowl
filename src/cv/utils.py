import json
import requests
import urllib

def get_auth_RTMP_url(
  rtmpUrl,
  loginUrl,
  rtmpRequestUrl,
  username,
  password):
  """Authenticate to loginUrl, append authorization parameters to rtmpUrl"""
  print(username, password)
  body = {"username": username, "password": password}
  loginResponse = requests.post(loginUrl, data=body)
  if not loginResponse.ok:
    raise Exception(f"Auth server response not ok : {loginResponse.text}")
  connectsidCookie = {
    "connect.sid": loginResponse.cookies["connect.sid"]}
  rtmpRequestResponse = requests.post(
    rtmpRequestUrl, cookies=connectsidCookie)
  if not rtmpRequestResponse.ok:
    raise Exception(f"Auth server response not ok : {rtmpRequestResponse.text}")
  rtmpCreds = json.loads(rtmpRequestResponse.text)
  return rtmpUrl + "?" + urllib.parse.urlencode(rtmpCreds)