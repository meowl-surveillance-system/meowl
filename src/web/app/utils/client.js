"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const cassandra_driver_1 = __importDefault(require("cassandra-driver"));
exports.client = new cassandra_driver_1.default.Client({
    contactPoints: ['127.0.0.1'],
    localDataCenter: 'datacenter1',
    keyspace: 'streams',
});
//# sourceMappingURL=client.js.map