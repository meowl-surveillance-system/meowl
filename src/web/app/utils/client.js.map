{"version":3,"file":"client.js","sourceRoot":"","sources":["../../../app/utils/client.ts"],"names":[],"mappings":";;;;;AAAA,wEAAyC;AAE5B,QAAA,MAAM,GAAG,IAAI,0BAAS,CAAC,MAAM,CAAC;IACzC,aAAa,EAAE,CAAC,WAAW,CAAC;IAC5B,eAAe,EAAE,aAAa;IAC9B,QAAQ,EAAE,SAAS;CACpB,CAAC,CAAC"}