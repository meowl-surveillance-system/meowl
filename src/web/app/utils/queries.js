"use strict";
// Query names take the following form: <ACTION>_<TABLE NAME>_<COLUMNS YOU WANT
// TO DO THE ACTION ON>
Object.defineProperty(exports, "__esModule", { value: true });
exports.INSERT_USERSID = 'INSERT INTO users_id (user_id, email, username, password, sid) VALUES (?, ?, ?, ?, ?)';
exports.INSERT_USERSNAME = 'INSERT INTO users_name (user_id, email, username, password, sid) VALUES (?, ?, ?, ?, ?)';
exports.SELECT_USERSID_USERID_PASSWORD = 'SELECT user_id, password FROM users_name WHERE username = ?';
exports.UPDATE_USERSID_SID = 'UPDATE users_id SET sid = ? WHERE user_id = ?';
exports.UPDATE_USERSNAME_SID = 'UPDATE users_name SET sid = ? WHERE username = ?';
exports.SELECT_USERSNAME_SID = 'SELECT sid FROM users_id WHERE user_id = ?';
exports.SELECT_SID_SESSION = 'SELECT * FROM sessions WHERE sid = ?';
exports.SELECT_USERSNAME_USERID = 'SELECT user_id FROM users_name WHERE username = ?';
exports.SELECT_CAMERAID_USERID = 'SELECT camera_id FROM user_cameras WHERE user_id = ?';
exports.SELECT_USERID_CAMERAID = 'SELECT user_id FROM camera_users WHERE camera_id = ?';
exports.INSERT_CAMERAID_USERID = 'INSERT INTO user_cameras (user_id, camera_id) VALUES (?, ?)';
exports.INSERT_USERID_CAMERAID = 'INSERT INTO camera_users (user_id, camera_id) VALUES (?, ?)';
exports.INSERT_CAMERAID_STREAMID = 'INSERT INTO camera_streams (camera_id, stream_id, stream_date) VALUES(?, ?, ?)';
exports.SELECT_CAMERAID_STREAMID = 'SELECT stream_id FROM camera_streams WHERE camera_id = ?';
exports.SELECT_STREAMID_METADATA = 'SELECT stream_id FROM metadata WHERE stream_id = ?';
exports.SELECT_CAMERAID_STREAMID_SINGLE = 'SELECT stream_id FROM camera_streams WHERE camera_id = ? LIMIT 1';
exports.INSERT_CAMERAID_LIVE = 'INSERT INTO live_cameras (camera_id, live) VALUES (?, ?)';
exports.SELECT_CAMERAID = 'SELECT DISTINCT camera_id FROM camera_streams';
exports.SELECT_LIVE_CAMERAID = 'SELECT live FROM live_cameras WHERE camera_id = ?';
//# sourceMappingURL=queries.js.map