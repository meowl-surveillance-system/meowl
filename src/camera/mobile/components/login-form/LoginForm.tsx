import React from 'react';
import { Modal, Button, Alert } from 'react-native';
import { Text, Input } from 'react-native-elements';
import { v4 as uuidv4 } from 'uuid';
import AsyncStorage from '@react-native-community/async-storage';
import CookieManager from '@react-native-community/cookies';
import { login } from './../../utils/utils';

/**
 * A form for logging in and getting rtmp link
 */

interface LoginFormState {
  username: string;
  password: string;
  cameraId: string;
}

/**
 * A form to log into the request server and get credentials for streaming
 */
class LoginForm extends React.Component<LoginFormProps, LoginFormState> {
  constructor(props: LoginFormProps) {
    super(props);
    this.state = {
      username: 'wack',
      password: 'boi',
      cameraId: '',
    };
  }

  /**
   * Creates a new camera ID for this app if it doesn't exists
   */
  async componentDidMount() {
    const cameraId: string | null = await AsyncStorage.getItem('meowlCameraId');
    if (cameraId) {
      this.setState({ cameraId });
    } else {
      const newCameraId: string = uuidv4();
      this.setState({ cameraId: newCameraId });
      await AsyncStorage.setItem('meowlCameraId', newCameraId);
    }
    await CookieManager.clearAll();
  }

  /**
   * Handles the login response after the login attempt
   */
  async handleLogin() {
    const loginResponse: Response | null = await login(
      {
        username: this.state.username,
        password: this.state.password,
      },
      this.props.requestServerUrl,
    );
    if (loginResponse) {
      if (loginResponse.ok) {
        return loginResponse;
      } else {
        Alert.alert('Wrong username or password.');
        return null;
      }
    } else {
      Alert.alert('Wrong IP/Port Number. Please specify http:// or https://');
      return null;
    }
  }

  /**
   * Retrieve credentials for RTMP streaming
   */
  async retrieveRtmpCredentials() {
    try {
      const rtmpResponse = await fetch(
        this.props.requestServerUrl + '/auth/rtmpRequest',
        {
          method: 'POST',
          headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
            credentials: 'include',
          },
        },
      );
      if (rtmpResponse.ok) {
        const responseBody = await rtmpResponse.json();
        return responseBody;
      } else {
        Alert.alert('Incorrect cookie credentials from logging in.');
        return null;
      }
    } catch (error) {
      Alert.alert('Error:', error.toString());
      return null;
    }
  }

  /**
   * Logins the user into the server and get credentials for streaming
   */
  async onSubmit() {
    const loginResponse: Response | null = await this.handleLogin();
    if (loginResponse) {
      const rtmpCredentials = await this.retrieveRtmpCredentials();
      if (rtmpCredentials) {
        this.updateProps({
          isLoggedIn: true,
          userId: rtmpCredentials['userId'],
          // TODO(yliu): Change this to sessionId after server changes this minor inconsistency
          sessionId: rtmpCredentials['sessionID'],
          cameraId: this.state.cameraId,
        });
        this.setState({
          username: '',
          password: '',
        });
        return true;
      } else {
        return false;
      }
    } else {
      return false;
    }
  }

  /**
   * Updates the main state of outer components
   * @param props - Property to be updated
   */
  updateProps(props: object) {
    this.props.updateProps(props);
  }

  /**
   * Clears all privacy information from this form
   */
  componentWillUnmount() {
    this.setState({
      username: '',
      password: '',
    });
  }

  /**
   * Renders SettingsForm component with inputs that updates the overall App state
   */
  render() {
    return (
      <Modal
        animationType="slide"
        transparent={false}
        visible={!this.props.isLoggedIn}>
        <Text style={{ fontSize: 22 }}>Request Server IP & Port Number:</Text>
        <Input
          onChangeText={requestServerUrl =>
            this.updateProps({ requestServerUrl })
          }
          value={this.props.requestServerUrl}
        />
        <Text style={{ fontSize: 22 }}>Username:</Text>
        <Input
          onChangeText={username => this.setState({ username })}
          value={this.state.username}
        />
        <Text style={{ fontSize: 22 }}>Password:</Text>
        <Input
          secureTextEntry={true}
          onChangeText={password => this.setState({ password })}
          value={this.state.password}
        />
        <Button onPress={() => this.onSubmit()} title="Submit" />
      </Modal>
    );
  }
}

export default LoginForm;
