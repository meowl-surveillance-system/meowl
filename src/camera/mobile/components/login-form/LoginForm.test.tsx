import 'react-native';
import { Alert } from 'react-native';
import React from 'react';
import LoginForm from './LoginForm';

import renderer from 'react-test-renderer';
import * as utils from '../../utils/utils';

describe('LoginForm Component', () => {
  let loginFormInstance: LoginForm;
  let updatePropsMock: jest.Mock;

  beforeEach(() => {
    updatePropsMock = jest.fn();
    loginFormInstance = new LoginForm({
      requestServerUrl: '',
      rtmpServerUrl: '',
      isLoggedIn: false,
      updateProps: updatePropsMock,
    });
  });
  it('renders correctly', () => {
    const container = renderer
      .create(
        <LoginForm
          requestServerUrl=""
          rtmpServerUrl=""
          isLoggedIn={true}
          updateProps={(obj: object) => { }}
        />,
      )
      .toJSON();
    expect(container).toMatchSnapshot();
  });

  it('should call handleLogin and retrieveRtmpCredentials in a successful onSubmit', async () => {
    const handleLoginSpy = jest
      .spyOn(loginFormInstance, 'handleLogin')
      .mockResolvedValue(
        new Response(
          {},
          {
            status: 200,
            statusText: 'yes',
            headers: {
              'Content-type': 'application/json',
            },
          },
        ),
      );
    const retrieveRtmpCredentialsSpy = jest.spyOn(
      loginFormInstance,
      'retrieveRtmpCredentials',
    );
    await loginFormInstance.onSubmit();
    expect(handleLoginSpy).toHaveBeenCalled();
    expect(retrieveRtmpCredentialsSpy).toHaveBeenCalled();
  });

  it('should call updateProps on a succesfully login', async () => {
    const handleLoginSpy = jest
      .spyOn(loginFormInstance, 'handleLogin')
      .mockResolvedValue(
        new Response(
          {},
          {
            status: 200,
            statusText: 'yes',
            headers: {
              'Content-type': 'application/json',
            },
          },
        ),
      );
    const retrieveRtmpCredentialsSpy = jest
      .spyOn(loginFormInstance, 'retrieveRtmpCredentials')
      .mockResolvedValue({
        userId: 'hello',
        // TODO(yliu): fix this sessionID to sessionId
        sessionID: 'world',
        isLoggedIn: true,
      });
    await loginFormInstance.onSubmit();
    expect(updatePropsMock).toHaveBeenCalledWith({
      cameraId: '',
      userId: 'hello',
      sessionId: 'world',
      isLoggedIn: true,
    });
  });

  it('should call login method in handleLogin', async () => {
    const loginSpy: jest.SpyInstance = jest.spyOn(utils, 'login');
    await loginFormInstance.handleLogin();
    expect(loginSpy).toHaveBeenCalled();
  });

  it('should alert the user on unsuccessful login', async () => {
    const loginSpy: jest.SpyInstance = jest
      .spyOn(utils, 'login')
      .mockResolvedValue(null);
    const alertSpy: jest.SpyInstance = jest.spyOn(Alert, 'alert');
    await loginFormInstance.handleLogin();
    expect(alertSpy).toHaveBeenCalled();
  });
});
