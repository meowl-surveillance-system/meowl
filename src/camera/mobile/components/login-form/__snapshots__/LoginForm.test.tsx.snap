// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Component renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={false}
  visible={false}
>
  <Text
    style={
      Object {
        "fontSize": 22,
      }
    }
    theme={
      Object {
        "colors": Object {
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "primary": "#2196f3",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "ios": Object {
              "error": "#ff3b30",
              "primary": "#007aff",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#8F0CE8",
          "success": "#52c41a",
          "warning": "#faad14",
        },
      }
    }
  >
    Request Server IP & Port Number:
  </Text>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        editable={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "fontSize": 22,
      }
    }
    theme={
      Object {
        "colors": Object {
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "primary": "#2196f3",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "ios": Object {
              "error": "#ff3b30",
              "primary": "#007aff",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#8F0CE8",
          "success": "#52c41a",
          "warning": "#faad14",
        },
      }
    }
  >
    Username:
  </Text>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        editable={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "fontSize": 22,
      }
    }
    theme={
      Object {
        "colors": Object {
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "primary": "#2196f3",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "ios": Object {
              "error": "#ff3b30",
              "primary": "#007aff",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#8F0CE8",
          "success": "#52c41a",
          "warning": "#faad14",
        },
      }
    }
  >
    Password:
  </Text>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        editable={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Submit
      </Text>
    </View>
  </View>
</Modal>
`;
